# Generated by Django 3.2.3 on 2021-06-07 13:16

from django.conf import settings
import django.core.validators
from django.db import migrations, models
import django.db.models.deletion
import fulfillment_service.utils.fields
import uuid


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='Wallet',
            fields=[
                ('id', models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False)),
                ('bank_name', models.CharField(max_length=50)),
                ('card_number', models.CharField(max_length=19, unique=True, validators=[django.core.validators.RegexValidator(message='Invalid card number', regex='^[0-9]{12,19}$')])),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='Transaction',
            fields=[
                ('id', models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False)),
                ('name', models.CharField(max_length=100)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('sum', fulfillment_service.utils.fields.BigDecimalField(decimal_places=2, max_digits=17)),
                ('wallet', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='wallets.wallet')),
            ],
            options={
                'ordering': ('created_at',),
            },
        ),
    ]
